lib/tasks/javascript.rake

```diff
@@ -16,23 +16,55 @@ def library_src
   "#{Rails.root}/node_modules"
 end
 
-def html_for_section(group)
-  icons =
-    group["icons"].map do |icon|
-      class_attr = icon["diversity"] ? " class=\"diversity\"" : ""
-      "    {{replace-emoji \":#{icon["name"]}:\" (hash lazy=true#{class_attr} tabIndex=\"0\")}}"
+def gjs_for_emoji_sections(groups_json)
+  constants =
+    groups_json.map do |group|
+      section_name = group["name"]
+
+      js = +"    { name: \"#{section_name}\", icons: ["
+
+      group["icons"].each do |icon|
+        js += "      ['#{icon["name"]}',"
+        js += icon["diversity"] ? "true]," : "],"
+      end
+
+      js += "    ] }"
     end
 
-  <<~HTML
-    <div class="section" data-section="#{group["name"]}">
-      <div class="section-header">
-        <span class="title">{{i18n "emoji_picker.#{group["name"]}"}}</span>
-      </div>
-      <div class="section-group">
-        #{icons.join("\n").strip}
-      </div>
-    </div>
-  HTML
+  <<~GJS
+    import Component from "@glimmer/component";
+    import { concat, get, hash } from "@ember/helper";
+    import replaceEmoji from "discourse/helpers/replace-emoji";
+    import i18n from "discourse-common/helpers/i18n";
+
+    const EMOJI_GROUP_SECTIONS = [
+      #{constants.join(",\n")}
+    ]
+
+    export default class EmojiGroupSections extends Component {
+      emojiGroupSections = EMOJI_GROUP_SECTIONS;
```

Comment:
Had to add this just to avoid https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/no-empty-glimmer-component-classes.md.

Reply:
The linting rule is telling us to make it a template-only component like this:

```
export default <template>
  ...
</template>
```

Which would be slightly faster to render (since there is no class to initialize)